src/main/java/org/shipkit/auto/version/AutoVersionPlugin.java

```diff
@@ -3,14 +3,19 @@
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
 
+
 /**
  * The plugin, ideally with zero business logic, but only the Gradle integration code
  */
 public class AutoVersionPlugin implements Plugin<Project> {
 
-    public void apply(Project project) {
-        DeductedVersion version = new AutoVersion(project.getProjectDir()).deductVersion();
-        project.allprojects(p -> p.setVersion(version.getVersion()));
-        project.getExtensions().getExtraProperties().set("shipkit-auto-version.previous-version", version.getPreviousVersion());
+  public void apply(Project project) {
+    DeductedVersion version = new AutoVersion(project.getProjectDir()).deductVersion();
+    if (project.getVersion().equals(Project.DEFAULT_VERSION)) {
```

Comment:
I'd love to see unit test for this new conditional complexity.

Reply:
Done